llvm/lib/SYCLLowerIR/SYCLConditionalCallOnDevice.cpp

```diff
@@ -0,0 +1,108 @@
+//===-- SYCLConditionalCallOnDevice.cpp - SYCLConditionalCallOnDevice Pass
+//--===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Pass performs transformations on functions which represent the conditional
+// call to application's callable object. The conditional call is based on the
+// SYCL device's aspects or architecture passed to the functions.
+//
+//===----------------------------------------------------------------------===//
+
+#include "llvm/SYCLLowerIR/SYCLConditionalCallOnDevice.h"
+
+#include "llvm/IR/Function.h"
+#include "llvm/IR/Instructions.h"
+
+using namespace llvm;
+
+PreservedAnalyses
+SYCLConditionalCallOnDevicePass::run(Module &M, ModuleAnalysisManager &) {
+  // find call_if_on_device_conditionally function
+  SmallVector<Function *, 4> FCallers;
+  for (Function &F : M.functions()) {
+    if (F.getName().contains("call_if_on_device_conditionally_helper") ||
+        !F.getName().contains("call_if_on_device_conditionally"))
+      continue;
+
+    FCallers.push_back(&F);
+  }
+
+  int FCallerIndex = 0;
+  for (Function *FCaller : FCallers) {
+    // Find call to @CallableXXX in call_if_on_device_conditionally function
+    // (FAction). FAction should be a literal (i.e. not a pointer)
+    Function *FAction = nullptr;
+    for (BasicBlock &BB : *FCaller) {
+      for (Instruction &I : BB) {
+        if (CallInst *callInst = dyn_cast<CallInst>(&I)) {
+          Value *calledValue = callInst->getCalledOperand();
+          FAction = dyn_cast<Function>(calledValue);
+        }
+      }
+    }
+
+    if (!FAction)
+      continue;
+
+    // Create a new function type with an additional function pointer argument
+    std::vector<Type *> NewParamTypes;
+    Type *FActionType = FAction->getType();
+    NewParamTypes.push_back(
+        PointerType::getUnqual(FActionType)); // Add function pointer to FAction
+    FunctionType *OldFCallerType = FCaller->getFunctionType();
+    for (Type *Ty : OldFCallerType->params()) {
+      NewParamTypes.push_back(Ty);
+    }
+
+    FunctionType *NewFCallerType =
```

Comment:
```suggestion
    auto *NewFCallerType =
```

Reply:
This one was not applied

Reply:
Oh, I'm sorry, instead I changed to `auto` [here](https://github.com/intel/llvm/pull/14228/commits/70423991f8b2fc4e43bebe17befdb1258069ebcf#diff-b6be3db7e1038f2c558c1f19ddce021c6b22e2e2a960714783c05496bd82abe9L87-R99) and thought that I fixed the original comment😂